spk/tvheadend/Makefile

```diff
@@ -9,29 +9,38 @@ DEPENDS = cross/busybox cross/$(SPK_NAME)
 
 MAINTAINER = m4tt075
 DESCRIPTION = Tvheadend is a TV streaming server and recorder for Linux, FreeBSD and Android supporting DVB-S, DVB-S2, DVB-C, DVB-T, ATSC, ISDB-T, IPTV, SAT IP and HDHomeRun as input sources. Tvheadend offers HTTP, HTSP and SAT IP streaming.
-ADMIN_PORT = 9981
 RELOAD_UI = yes
 DISPLAY_NAME = Tvheadend
-CHANGELOG = "Update to Tvheadend 4.2.4 (Release), including update of libhdhomerun to 20170930"
-
+STARTABLE = yes
+CHANGELOG = "1. Update TVH to 4.2.5 (maintenance release), 2. Use DSM 5+6 generic service"
 HOMEPAGE = https://www.lonelycoder.com/tvheadend/
-LICENSE  = GPL v3
+LICENSE = GPL v3
 
-INSTALLER_SCRIPT = src/installer.sh
-SSS_SCRIPT       = src/dsm-control.sh
-FWPORTS          = src/${SPK_NAME}.sc
-CONF_DIR         = src/conf/
-WIZARDS_DIR      = src/wizard/
+WIZARDS_DIR = src/wizard/
+CONF_DIR = src/conf/
 
-INSTALL_PREFIX = /usr/local/$(SPK_NAME)
+# 'auto' reserved value grabs SPK_NAME
+SERVICE_USER = auto
+SERVICE_SETUP = src/service-setup.sh
 
+# Admin link
+ADMIN_PORT = 9981
+
+PRE_COPY_TARGET = tvheadend_install
 POST_STRIP_TARGET = tvheadend_extra_install
 
 BUSYBOX_CONFIG = usrmng
 ENV += BUSYBOX_CONFIG="$(BUSYBOX_CONFIG)"
 
 include ../../mk/spksrc.spk.mk
 
+.PHONY: tvheadend_install
+# Replace standard copy/install targets
+tvheadend_install:
+	rm -fr $(STAGING_DIR)
+	mkdir $(STAGING_DIR)
+	mkdir --parents $(STAGING_INSTALL_PREFIX)
+
 .PHONY: tvheadend_extra_install
 tvheadend_extra_install:
```

Comment:
I really wonder if this target is really usefull too... With `DSM_UI_DIR`, folder is copied without such install command.